ksql-engine/src/test/resources/query-validation-tests/README.md

```diff
@@ -209,6 +207,129 @@ The expected exception can define the following attributes:
 | type      | (Optional) The fully qualifid class name of the _exact_ exception |
 | message   | (Optional) The full text of the message within the exception, i.e. the error message that would be displayed to the user |
 
+### Post Conditions
+A test can define a set of post conditions that must be met for the test to pass, e.g.
+
+```json
+{
+  "post": {
+    "sources": [
+      {"name": "INPUT", "type": "stream", "keyField": null}
+    ]
+  }
+}
+```
+
+Post conditions current support the following checks:
+
+| Attribute | Description |
+|-----------|:------------|
+| sources   | (Optional) A list of sources that must exist in the metastore after the statements have executed. This list does not need to define every source. |
+
+#### Sources
+A post condition can define the list of sources that must exist in the metastore. A source might be:
+
+```json
+{
+  "name": "S1",
+  "type": "table",
+  "keyField": {"name": "FOO", "legacyName": "KSQL_INTERNAL_COL_0", "legacySchema": {"type": "STRING"}},
+  "valueSchema": {"type": "STRUCT", "fields": [
```

Comment:
I feel like it would be simpler to just have our usual SQL schema parser handle this - why make it explicit in JSON format?

Reply:
Good idea.